File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/preferences/BasePreferences.java
package com.hgianastasio.biblioulbrav2.data.disk.preferences;

import android.content.Context;
import android.content.SharedPreferences;


/**
 * Created by heitorgianastasio on 4/28/17.
 */

public abstract class BasePreferences {
    private Context mContext;
    private SharedPreferences preferences;
    public static final String BIBLIOULBRA_PREFERENCE_KEY = "biblioulbra.preferences";

    public BasePreferences(Context mContext) {
        this.mContext = mContext;
        preferences = mContext.getSharedPreferences(BIBLIOULBRA_PREFERENCE_KEY,Context.MODE_PRIVATE);
    }

    protected Context getContext() {
        return mContext;
    }

    protected SharedPreferences getPreferences() {
        return preferences;
    }
}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/preferences/BasePreferences.kt
package com.hgianastasio.biblioulbrav2.data.disk.preferences

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by heitorgianastasio on 4/28/17.
 */
abstract class BasePreferences(protected val context: Context) {
    protected val preferences: SharedPreferences

    companion object {
        const val BIBLIOULBRA_PREFERENCE_KEY = "biblioulbra.preferences"
    }

    init {
        preferences = context.getSharedPreferences(BIBLIOULBRA_PREFERENCE_KEY, Context.MODE_PRIVATE)
    }
}